File: src/main/java/com/example/myproject/entity/ActionList.java
package com.example.myproject.entity;


import io.smallrye.common.constraint.NotNull;
import jakarta.persistence.*;
import lombok.Data;

import java.io.Serializable;

@Data
@Entity
@Table(name = "actionlist")
public class ActionList implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "actcode")
    private String actCode;
    
    @Column
    private String investors;
    
    @Column(name = "create_date")
    private String createDate;
    
    @Column(name = "project_scope")
    private String projectScope;
    
    @Column
    private String standards;
    
    @Column(name = "schedule_start_date")
    private String scheduleStartDate;
    
    @Column(name = "schedule_end_date")
    private String scheduleEndDate;
    
    @Column
    private String position;
    
    @Column
    private String status;
    
    @Column(name = "detail_information")
    private String detailInformation;
    
    @Column
    private  String risk;
    
    @Column(name = "action_cycle")
    private  String actionCycle;
    
    @Column(name = "department_collaboration")
    private String departmentCollaboration;
    
    @Column(name = "assessment_conclusion")
    private String assessmentConclusion;
    
    @Column(name = "corrective_measures")
    private String correctiveMeasures;
    
    
    
}

File: Graduation_Project/front end/GP_project/src/api/actionlist.ts
import request from "@/utils/request";

export type Common<T> = {
    count:string
    code: string
    data:T
    mesg:string
}
//actionlist表单
export type actionItem = {
    type:string
    actCode:string
    investors:string
    createDate:string
    projectScope:string
    standards:string
    status:string
    position:string
    detailInformation:string
    risk:string
    actionCycle:string
    departmentCollaboration:string
    assessmentConclusion:string
    correctiveMeasures:string
    scheduleStartDate:string
    scheduleEndDate:string
    id:number
}
//新造的表单
export type CreateOrEditActItem = Omit<actionItem,
    'assessmentConclusion' | 'correctiveMeasures'> &{id:number}
    

export const getActionList = (page:number) =>{
    return request<{code:number;count:number;msg:string;data:actionItem[]}>({
        method:"GET",
        url:`/actionlist/list?page=${page}&size=15`
    })
}
//添加表单和更新表单
export const saveOrUpdateAction = (actinfo :actionItem | CreateOrEditActItem) =>{
    return request<{code:number;msg:string;data:actionItem[]}>({
        method:"PUT",
        url:"/actionlist/",
        data:actinfo
    })
}
//根据id删除
export const deleteAction = (id:number) =>{
    return request<{code:number;msg:string;data:actionItem[]}>({
        method:"DELETE",
        url:`/actionlist/${id}`
    })
}
//根据id查找
export const getActionById = (id:String) =>{
    return request<{code:number;msg:string;data:actionItem}>({
        method:"GET",
        url:`/actionlist/${id}`,
        params:{
            id,
        }
    })
}
